enum.go.tmpl

```diff
@@ -31,20 +31,12 @@ func (x {{$name}}) String() string {
 	return {{$name}}_name[{{$type}}(x)]
 }
 
-func (x {{$name}}) MarshalJSON() ([]byte, error) {
-	buf := bytes.NewBufferString(`"`)
-	buf.WriteString({{$name}}_name[{{$type}}(x)])
-	buf.WriteString(`"`)
-	return buf.Bytes(), nil
+func (x Status) MarshalText() ([]byte, error) {
+	return []byte({{$name}}_name[{{$type}}(x)]), nil
 }
 
-func (x *{{$name}}) UnmarshalJSON(b []byte) error {
-	var j string
-	err := json.Unmarshal(b, &j)
-	if err != nil {
-		return err
-	}
-	*x = {{$name}}({{$name}}_value[j])
+func (x *{{$name}}) UnmarshalText(b []byte) error {
+	*x = {{$name}}({{$name}}_value[string(b)])
 	return nil
 }
```

Comment:
This PR doesn't change the ENUM's behavior.  But perhaps we should document that on unknown values, the unmarshaler defaults to `0` zero value.